usaspending_api/etl/swap_in_new_table_strategies.py

```diff
@@ -0,0 +1,955 @@
+                        new_view_name=dep_view["dep_view_name"],
+                    ),
+                ]
+                self.logger.info(f"Running view swap SQL in an atomic transaction:\n{pformat(view_rename_sql)}")
+                cursor.execute("\n".join(view_rename_sql))
+
+    def _validate_options(self) -> None:
+        """Validates the combination of options provided to this command. Will
+        raise an exception if any are invalid
+        """
+        is_undo = self._options["undo"]
+        keep_old_data = self._options["keep-old-data"]
+        foreign_key = self._options["allow-foreign-key"]
+        if is_undo is not None or keep_old_data is not None or foreign_key is not None:
+            raise ArgumentError(
+                f"The swap in new table command is using the IncrementalLoadSwapInTableStrategy strategy class. IncrementalLoadSwapInTableStrategy does not support these options '--keep-old-data', '--undo', and '--allow-foreign-key'."
+            )
+
+    def _validate_live_tables(self, cursor, table: str):
+        """Checks various things that need to be true before this strategy can work.
+
+        Args:
+            table: The live table name to validate.
+        """
+        table = self._options["table"]
+        validation_query = f"""
+            SELECT EXISTS (
+            SELECT FROM pg_catalog.pg_class c
+            JOIN   pg_catalog.pg_namespace n ON n.oid = c.relnamespace
+            WHERE  c.relname = '{table}'
+            AND    c.relkind = 'r'    -- only tables
+            );
+        """
+        cursor.execute(validation_query)
+        validation_results = ordered_dictionary_fetcher(cursor)
+        if len(validation_results) == 0 or validation_results[0]["exists"] is False:
+            raise ValueError(f"Swap in new table command cannot proceed because {table} does not exist.")
+
+    def _validate_temp_tables(self, cursor, temp_table_list: List):
+        """Checks various things that need to be true before this strategy can work.
+
+        Args:
+            temp_table_list: A list of temp tables that need validation.
+        """
+        for temp_table_name in temp_table_list:
+            validation_query = f"""
+                SELECT EXISTS (
+                SELECT FROM pg_catalog.pg_class c
+                JOIN   pg_catalog.pg_namespace n ON n.oid = c.relnamespace
+                WHERE  c.relname = {temp_table_name}
+                AND    c.relkind = 'r'
+                );
+            """
+            cursor.execute(validation_query)
+            validation_results = ordered_dictionary_fetcher(cursor)
+            if len(validation_results) == 0 or validation_results[0]["exists"] is False:
+                raise ValueError(f"Swap in new table command cannot proceed because {temp_table_name} does not exist.")
+
+    def _execute_upserts(self, cursor, qualified_dest_table, qualified_upsert_postgres_table):
```

Comment:
There are a few issues with this method:
- cursor is executing itself rather than the prepared sql
- columns are still hardcoded to `AWARD_SEARCH_COLUMNS` (although this may have been a temporary workaround with the `pyspark` blocker)
- Not sure why you're overwriting the last three columns on the `INSERT` with timestamps and a `NULL`. The last three columns of `AWARD_SEARCH_COLUMNS` are `total_iija_outlay`, `total_iija_obligation`, and `total_outlays`. 
- I believe the `INSERT` statement will need table qualifiers on both the `INSERT` and `VALUES` lists.

Reply:
> cursor is executing itself rather than the prepared sql

Can you show me an example, my `cursor.execute` code matches other code.

Reply:
> columns are still hardcoded to AWARD_SEARCH_COLUMNS (although this may have been a temporary workaround with the pyspark blocker)

Why is this an issue? I used AWARD_SEARCH_COLUMNS variable based off how other parts of our code base do it.

Reply:
> Not sure why you're overwriting the last three columns on the INSERT with timestamps and a NULL. The last three columns of AWARD_SEARCH_COLUMNS are total_iija_outlay, total_iija_obligation, and total_outlays.

Again this was based off other existing code in our code base. Let me take a deeper look.

Reply:
I've addressed
* I believe the INSERT statement will need table qualifiers on both the INSERT and VALUES lists.
* cursor is executing itself rather than the prepared sql

Reply:
In other parts of the code, we pull things like `AWARD_SEARCH_COLUMNS` out of the `TABLE_SPEC`. In the current state, the `_execute_upeserts()` method is hardcoded to only work for one table

Reply:
Oh gotcha, ok

Reply:
addressed everything in this thread